### ISSUE TITLE
Bug: Terraform cannot remove authorized_networks config from a google_sql_database_instance resource

### ISSUE THREAD

**Issue by crystalthenR2R** [hashicorp/terraform-provider-google#17284]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team)
  to expedite investigation and resolution of this issue.

### Terraform Version

```
Terraform v1.7.3
on linux_arm64
+ provider registry.terraform.io/hashicorp/google v5.16.0
```

### Affected Resource(s)

google_sql_database_instance

### Terraform Configuration

Bug appears after applying Config 1 then config 2 -

Configuration 1:
```tf
terraform {
  backend "gcs" {
    bucket = "***"
    prefix = "***"
  }
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "= 5.16.0"
    }
  }
}

provider "google" {
  project = "***"
  region  = "europe-west1"
}

resource "google_sql_database_instance" "instance" {
  name             = "test-instance"
  database_version = "MYSQL_8_0_31"
  deletion_protection = false

  settings {
    tier      = "db-custom-2-8192"
    disk_size = 10

    ip_configuration {
      ipv4_enabled = true

      authorized_networks {
        name = "test"
        value = "1.1.1.1/32"
      }
    }
  }
}
```

Configuration 2:
```tf
terraform {
  backend "gcs" {
    bucket = "***"
    prefix = "***"
  }
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "= 5.16.0"
    }
  }
}

provider "google" {
  project = "***"
  region  = "europe-west1"
}

resource "google_sql_database_instance" "instance" {
  name             = "test-instance"
  database_version = "MYSQL_8_0_31"
  deletion_protection = false

  settings {
    tier      = "db-custom-2-8192"
    disk_size = 10
  }
}
```

### Debug Output

https://gist.github.com/crystalthenR2R/7e5737020d1f3c80914696ef0f32f251

### Expected Behavior

Authorized networks should be removed from the Cloud SQL instance if the Terraform configuration has no `authorized_networks` block.

### Actual Behavior

Terraform returns `No changes. Your infrastructure matches the configuration.`

### Steps to reproduce

1. `terraform apply` configuration 1
2. `terraform plan` configuration 2

### Important Factoids

We're not seeing the bug consistently with all configs / TF runs.
e.g. a `plan` on someone's laptop will show the authorized network being deleted, while a `plan` with the same config and state file on a CI server will show **No Changes.**

For reference here's the OS information for the VM where I generated the above debug log:
```
# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.3 LTS
Release:        22.04
Codename:       jammy

# uname -a
Linux 9a96c20eb0aa 6.5.11-300.fc39.aarch64 #1 SMP PREEMPT_DYNAMIC Wed Nov  8 22:11:28 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux
```

**Comment by edwardmedia** [comment]
Yes, I repro. The diff can't be detected in the plan after `ip_configuration` is commented. We can update other attributes inside `ip_configuration` which match what `gcloud sql instances patch` offers.   But what properly behavior should be? It seems updating ip_configuration is supported by [the api](https://cloud.google.com/sql/docs/mysql/admin-api/rest/v1beta4/instances/patch)

**Comment by crystalthenR2R** [comment]
Thanks @edwardmedia for verifying.

It's not necessary to remove the whole `ip_configuration` block to see the bug.

In our code we're setting `ip_configuration` without `authorized_networks`, and somebody has manually added the authorized network in the console. When I run a TF plan/apply in this case from my laptop the config is removed. But a TF plan/apply from our CI server doesn't pick it up.

As mentioned it seems inconsistent between OSs so I couldn't get a simple example of the bug in this case.
